src/openzaak/utils/migrations.py

```diff
@@ -0,0 +1,68 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2022 Dimpact
+import logging
+from urllib.parse import urlsplit, urlunsplit
+
+from django.db.models.functions import Length
+
+logger = logging.getLogger(__name__)
+
+
+def get_service(model, url: str):
+    """
+    copy-paste from zgw_consumers.Service.get_service method, which can't be used
+    during migrations
+    """
+    split_url = urlsplit(url)
+    scheme_and_domain = urlunsplit(split_url[:2] + ("", "", ""))
+
+    candidates = (
+        model.objects.filter(api_root__startswith=scheme_and_domain)
+        .annotate(api_root_length=Length("api_root"))
+        .order_by("-api_root_length")
+    )
+
+    # select the one matching
+    for candidate in candidates.iterator():
+        if url.startswith(candidate.api_root):
+            return candidate
+
+    return None
+
+
+def fill_service_urls(
+    apps,
+    model,
+    url_field: str,
+    service_base_field: str,
+    service_relative_field: str,
+    fake_etag: bool = False,
+):
+    """
+    helper function to migrate from UrlField to ServiceUrlField
+
+    :param fake_etag: workaround to migrate ETagMixin models
+    """
+    Service = apps.get_model("zgw_consumers", "Service")
+    cache_get_service = {}
+
+    for instance in model.objects.exclude(**{url_field: ""}):
+        url = getattr(instance, url_field)
+
+        if url in cache_get_service:
```

Comment:
`@functools.lru_cache` on the `get_service` function should be a bit more straightforward and simplify this function.

Reply:
Thank you, I didn't know about it. But I've changed the implementation of the `fill_service_urls` function and now I don't want to cache the empty values of `get_service` function, so I kept my implementation of caching